core/src/main/scala/filodb.core/memstore/FiloSchedulers.scala

```diff
@@ -0,0 +1,19 @@
+package filodb.core.memstore
+
+import filodb.core.GlobalConfig
+
+object FiloSchedulers {
+  val IngestSchedName = "ingestion-shard"
+  val FlushSchedName = "flush-sched"
+  val IOSchedName = "filodb.io"
+  val QuerySchedName = "query-sched"
+  val PopulateChunksSched = "populate-odp-chunks"
+
+  def assertThreadName(name: String): Unit = {
+
+    if (GlobalConfig.systemConfig.getBoolean("filodb.scheduler.enable-assertions")) {
```

Comment:
Assign the config boolean to a object-level val, and then the check is compiled away at runtime. Do we expect to change the config setting dynamically?

Reply:
Good idea. done